students/navdeep/session06/mailroom_part4NoTest.py

```diff
@@ -0,0 +1,272 @@
+        elif user_donor_option == 'exit':
+            return
+        else:
+            print("Please select a valid option:")
+            print()
+
+def addNewDonor(donor_dict):
+    first_name = None
+    last_name = None
+    full_name = None
+    invalid = True
+    while invalid == True:
+        try:
+            #first_name = input("Enter first name for donor: ")
+            #last_name = input("Enter last name for donor: ")
+            first_name = "Kelly"
+            last_name = "Pratt"
+            first_name, last_name = first_name.strip(),last_name.strip()
+            if first_name == "" or last_name == "":
+                raise ValueError
+        except ValueError as e:
+            print("The name you entered is invalid. Enter proper first/last name")
+        else:
+            first_name, last_name = first_name.lower(), last_name.lower()
+            first_name,last_name = first_name.title(), last_name.title()
+            invalid = False
+    full_name = first_name + ' ' + last_name
+    #Return statement used for testing purposes
+    return full_name
+    #addNewDonation(donor_dict, full_name)
+
+def addNewDonation(donor_dict, new_donor):
+    """
+    Allows the user to input a new donation for the user. The user must
+    enter a valid amount (integer or float).  The user must
+    also enter a value greater than 0.  If the donor currently exists
+    in the dictionary, the donor's new donation will be appended to the list stored under
+    their name, which is the key in the dictionary. If the donor does
+    not exist in the dictionary, a new key will be created along with a new
+    list for that key.
+    :param donor_list: the dictionary of donors and their donation amounts
+    :param new_donor: the name of the new donor
+    """
+    try:
+        #new_donation = float(input("Enter a new donation amount: "))
+        new_donation = 1000
+        while new_donation < 0:
+            new_donation = float(input("Enter something greater than 0: "))
+    except ValueError as e:
+        print("The amount you entered is an invalid value.")
+    else:
+        if donorExists(donor_dict, new_donor):
+            donor_dict[new_donor].append(new_donation)
+        else:
+            donor_dict[new_donor.strip()] = [new_donation]
+        #Return used for testing purposes. Testing
+        #To see if dictionary updated with new donor
+        #and new donation
+        return donor_dict
+        #printThankYou(new_donor, new_donation)
+
+def donorExists(donor_dict, new_donor):
+    """
+    This function will test whether or not a donor already exists in the dictionary
+    :param donor_list: A dictionary of donor names and the values they have donated
+    :param new_donor: the name of the donor the user most recently inputted
+    :return: True if donor exists in the dictionary as a key, false otherwise
+    """
+    if new_donor in donor_dict:
+        return True
+    else:
+        return False
```

Comment:
You could also wrap this into the same function where you are adding a new user.. if it exists, just return the info, otherwise, add it to the database.

Reply:
Could this also be replaced with the setdefault function we talked about in class?